ouroboros-consensus-cardano/src/shelley/Ouroboros/Consensus/Shelley/Ledger/Inspect.hs

```diff
@@ -8,166 +8,55 @@
-        , protocolUpdateState = UpdateState {
-              proposalVotes         = votes
-            , proposalReachedQuorum = length votes >= fromIntegral quorum
-            }
-        }
-    | (proposal, votes) <- Map.toList $ invertMap proposals
-    ]
-  where
-    invertMap :: Ord b => Map a b -> Map b [a]
-    invertMap = Map.fromListWith (<>) . fmap swizzle . Map.toList
-      where
-        swizzle (a, b) = (b, [a])
-
-    -- Updated proposed within the proposal window
-    proposals :: Map (SL.KeyHash 'SL.Genesis (EraCrypto era)) (Core.PParamsUpdate era)
-    SL.ProposedPPUpdates proposals =
-          fromMaybe SL.emptyPPPUpdates
-        . Core.getProposedPPUpdates
-        . view SL.newEpochStateGovStateL
-        . shelleyLedgerState
-        $ st
-
-    -- A proposal is accepted if the number of votes is equal to or greater
-    -- than the quorum. The quorum itself must be strictly greater than half
-    -- the number of genesis keys, but we do not rely on that property here.
-    quorum :: Word64
-    quorum = SL.sgUpdateQuorum genesis
-
-    -- The proposals in 'SL.proposals' are for the upcoming epoch
-    -- (we ignore 'futureProposals')
-    currentEpoch :: EpochNo
-    currentEpoch = SL.nesEL . shelleyLedgerState $ st
-
-{-------------------------------------------------------------------------------
-  Inspection
--------------------------------------------------------------------------------}
-
 data ShelleyLedgerUpdate era =
-    ShelleyUpdatedProtocolUpdates [ProtocolUpdate era]
+  ShelleyUpdatedPParams
+    !(StrictMaybe (Core.PParams era))
+    !EpochNo
 
-deriving instance Eq (Core.PParamsUpdate era) => Eq (ShelleyLedgerUpdate era)
-deriving instance Show (Core.PParamsUpdate era) => Show (ShelleyLedgerUpdate era)
+deriving instance Eq (Core.PParams era) => Eq (ShelleyLedgerUpdate era)
+deriving instance Show (Core.PParams era) => Show (ShelleyLedgerUpdate era)
 
-instance Show (Core.PParamsUpdate era) => Condense (ShelleyLedgerUpdate era) where
+instance Show (Core.PParams era) => Condense (ShelleyLedgerUpdate era) where
   condense = show
 
 instance ShelleyBasedEra era => InspectLedger (ShelleyBlock proto era) where
   type LedgerWarning (ShelleyBlock proto era) = Void
   type LedgerUpdate  (ShelleyBlock proto era) = ShelleyLedgerUpdate era
 
-  inspectLedger tlc before after = do
+  inspectLedger _tlc before after = do
       guard $ updatesBefore /= updatesAfter
-      return $ LedgerUpdate $ ShelleyUpdatedProtocolUpdates updatesAfter
+      return $ LedgerUpdate updatesAfter
     where
-      genesis :: SL.ShelleyGenesis (EraCrypto era)
-      genesis = shelleyLedgerGenesis (configLedger tlc)
 
-      updatesBefore, updatesAfter :: [ProtocolUpdate era]
-      updatesBefore = protocolUpdates genesis before
-      updatesAfter  = protocolUpdates genesis after
+      updatesBefore, updatesAfter :: ShelleyLedgerUpdate era
+      updatesBefore = pparamsUpdate before
+      updatesAfter  = pparamsUpdate after
```

Comment:
FTR: With the new code, we will log one `LedgerUpdate` in every epoch (when we first tick across the Shelley voting deadline, ie 6k/f before the end of the epoch), even if nothing changed. This is in contrast to the previous state of the code. This seems fine to me, at least for now.